scipy/signal/tests/test_windows.py

```diff
@@ -696,10 +696,10 @@ def test_needs_params():
     for winstr in ['kaiser', 'ksr', 'gaussian', 'gauss', 'gss',
                    'general gaussian', 'general_gaussian',
                    'general gauss', 'general_gauss', 'ggs',
-                   'dss', 'dpss',
-                   'chebwin', 'cheb', 'exponential', 'poisson', 'tukey',
-                   'tuk', 'dpss']:
-        assert_raises(ValueError, get_window, winstr, 7)
+                   'dpss', 'chebwin', 'cheb', 'general_cosine',
```

Comment:
I fixed several issues:
1. There is not 'dss' window function.
2. "exponential", "poisson", "tukey", "tuk" does not need additional argument.
3. Missing window functions "general_cosine", "general_hamming" are added